simple_salesforce/login.py

```diff
@@ -37,12 +37,22 @@ def SalesforceLogin(
     * session -- Custom requests session, created in calling code. This
                  enables the use of requets Session features not otherwise
                  exposed by simple_salesforce.
+    * custom_url -- Supports Salesforce URL customization
     """
 
-    soap_url = 'https://{domain}.salesforce.com/services/Soap/u/{sf_version}'
-    domain = 'test' if sandbox else 'login'
+    soap_url = ""
+    domain = ""
 
-    soap_url = soap_url.format(domain=domain, sf_version=sf_version)
+    if custom_url:
+        soap_url = '{custom_url}/services/Soap/u/{sf_version}'
+        soap_url = soap_url.format(custom_url=custom_url, sf_version=sf_version)
+    else:
+        soap_url = 'https://'
+        soap_url += '{domain}.salesforce.com/services/Soap/u/'
+        soap_url += '{sf_version}'
+        domain = 'test' if sandbox else 'login'
+
+        soap_url = soap_url.format(domain=domain, sf_version=sf_version)
```

Comment:
This isn't optimized for readability IMO as it requires the user to know that the domain param will only be used if `not custom_url`. I think something like the following may be a little more readable:

```
if custom_url:
    soap_url = '{custom_url}/services/Soap/u/{sf_version}'.format(
        custom_url=custom_url, sf_version=sf_version
    )
else:
    soap_url = 'https://{domain}.salesforce.com/services/Soap/u/{sf_version}'.format(
        domain='test' if sandbox else 'login', sf_version=sf_version
    )
```